ckan/logic/__init__.py

```diff
@@ -674,7 +674,7 @@ def get_validator(validator):
         _validators_cache.update(validators)
         validators = _import_module_functions('ckan.logic.validators')
         _validators_cache.update(validators)
-        _validators_cache.update({'OneOf': formencode.validators.OneOf})
+        _validators_cache.update({'OneOf':'ckan.logic.validators.one_of'})
```

Comment:
You need the actual reference to the function, not a string. So `{'OneOf': ckan.logic.validators.one_of}`